docs/imagecustomizer/api/configuration/injectArtifactMetadata.md

```diff
@@ -0,0 +1,57 @@
+---
+parent: Configuration
+---
+
+# InjectArtifactMetadata type
+
+Defines a single artifact to be injected into a partition during image modification.
+
+This is used in the [`InjectFilesConfig`](./injectFilesConfig.md) array
+when performing injection via the [`inject-files` CLI command](../cli/inject-files.md).
+
+## Example
+
+```yaml
+partition:
+  idType: part-uuid
+  id: b9f59ced-d1a6-44a7-91d9-4d623a39b032
+destination: /EFI/BOOT/bootx64.efi
+source: ./bootx64.signed.efi
```

Comment:
Sorry if I missed your response earlier, how would a caller know what should be the source?

Reply:
It is my understanding that the user would get a yaml output out of the initial prism call and then supposed to sign bunch of input files. So is the expectation that caller would parse the produced yaml to figure out where are the unsigned binaries placed and where the signed binaries should be placed?

Reply:
The `output.artifacts` API will use fixed paths for the `source` and `destination`. So, a user will be able to assume the paths without needing to parse the YAML file. Though, if they want to parse the YAML file they can.

The `inject-files` API will be fully dynamic.

Reply:
More confused. How is output.artifacts static, given that kernel name includes the version?

And how would I know based on that where to place the signed binary? Based on the sample below, I dont have a good way without parsing the file:

- partition:
    idType: part-uuid
    id: b9f59ced-d1a6-44a7-91d9-4d623a39b032
  destination: /EFI/Linux/vmlinuz-6.6.51.1-5.azl3.efi
  source: ./vmlinuz-6.6.51.1-5.azl3.signed.efi
  unsignedSource: ./vmlinuz-6.6.51.1-5.azl3.unsigned.efi

Reply:
I guess it is more accurate to say that the pattern will be static. That is, `vmlinuz-<version>.signed.efi` and `vmlinuz-<version>.unsigned.efi`.

Reply:
I would vote for including the type.